files/image_config/config-setup/config-setup

```diff
@@ -107,7 +107,7 @@ reload_minigraph()
         echo "{}" > /etc/sonic/init_cfg.json
     fi
     sonic-db-cli CONFIG_DB FLUSHDB
-    sonic-cfggen -H -m -j /etc/sonic/init_cfg.json --write-to-db
+    sonic-cfggen -H -j /etc/sonic/init_cfg.json -m --write-to-db
```

Comment:
Yes, I double checked on DUT for the command sonic-cfggen. Order of `-m` and `-j` arguments does not make a difference.

Reply:
Looks like we need to fix that too...

Reply:
Will fix that.

Reply:
How do you intend to fix it?

Reply:
Can we change the src of sonic-cfggen to load `-j` init_cfg.json before `-m` minigraph?

Reply:
I believe we could reverse the order to load `-j` (json) files before we load the minigraph. I don't think it would have any other adverse effects, but it would require thorough testing.

Reply:
Joe, I tested the command `sonic-cfggen` as following: I copied the table from minigraph into init_cfg.json.
`    "DEVICE_NEIGHBOR_METADATA": {                                                                                                                                                           
        "ARISTA01T1": {
            "hwsku": "Arista-VM",
            "lo_addr": "None",
            "mgmt_addr": "172.16.141.104",
            "type": "LeafRouter"
        },
        "ARISTA02T1": {
            "hwsku": "Arista-VM",
            "lo_addr": "None",
            "mgmt_addr": "172.16.141.105",
            "type": "LeafRouter"
        },
        "ARISTA03T1": {
            "hwsku": "Arista-VM",
            "lo_addr": "None",
            "mgmt_addr": "172.16.141.106",
            "type": "LeafRouter"
        },
        "ARISTA04T1": {
            "hwsku": "Arista-VM",
            "lo_addr": "None",
            "mgmt_addr": "172.16.141.107",
            "type": "LeafRouter"
        }
    }
`
In init_cfg.json file, I just temporarily changed the values of key "type" from "LeafRouter" to "T0", "T1", "T2" and "T3". I ran the command `sonic-cfggen -m -f /etc/sonic/init_cfg.json --write-to-db` and then check the data in Config_DB. I found all the values of key "type" will be overwritten into "T0", "T1", "T2" and "T3". After that running another command `sonic-cfggen -f /etc/sonic/init_cfg.json -m --write-to-db`, the values of key "type" in Config_DB will also be overwritten into "T0", "T1", "T2" and "T3".

Reply:
Is the `-f` in your commands above a typo? Clearly, from the sonic-cfggen code I linked above, the order you pass the `-j` and `-m` flags shouldn't matter. So, to confirm, you proved that my assumption above is correct: that passing the `-m flag along with a `-j` flag will cause the minigraph to get loaded first followed be the JSON file(s), which wind up overwriting the minigraph data?

Reply:
This is a typo and should be `-j`.

Reply:
I have opened a PR to swap the order with which sonic-cfggen loads JSON files vs minigraph.xml here: https://github.com/Azure/sonic-buildimage/pull/4202. Now, all JSON files will be loaded *before* the minigraph.